linter.py

```diff
@@ -71,24 +111,50 @@ def split_match(self, match):
 
         if match:
             open_file_name = os.path.basename(self.view.file_name())
-            linted_file_name = match.group('file_name')
+            # Since the filename on the top row might be different than the open file if, for example,
+            # something is imported from another file. Use the filename from the footer is it's available.
+            linted_file_name = match.group('file_name_2') or match.group('file_name_1')
 
             if linted_file_name == open_file_name:
+
                 message_title = match.group('message_title')
                 message = match.group('message')
-                message_footer = match.group('message_footer') or ""
-
-                if message_title and message_title.strip():
-                    message = '"{0}"" {1} {2}'.format(
-                        message_title,
-                        message,
-                        message_footer
-                    )
-
-                line = max(int(match.group('line')) - 1, 0)
-                col = int(match.group('col')) - 1
+                message_footer = match.group('message_footer')
+                col = match.group('col_2') or match.group('col_1')
+                message_format = []
+
+                if message_title:
+                    message_format.append('[ {0} ]')
+                if message:
+                    message_format.append('{1}')
+                if message_footer:
+                    message_format.append('[ {2} ]')
+
+                message = " ".join(message_format).format(
+                    message_title,
+                    message,
+                    message_footer
+                )
+
+                # Get the start and ending indexes of the line and column
+                line_cols = col.replace(':', ',').split(',')
+                line_start = max(int(line_cols[0])-1, 0)
+                col_start = int(line_cols[1])
+                col_start -= 1
+
+                # Multi line error
+                if len(line_cols) == 4:
+                    line_end = max(int(line_cols[2])-1, 0)
+                    col_end = int(line_cols[3])
+                    near = " " * (self.view.text_point(line_end, col_end) - self.view.text_point(line_start, col_start))
+
+                # Single line error
+                else:
+                    col_end = int(line_cols[2])
+                    # Get the length of the column section for length of error
+                    near = " " * (col_end - col_start)
```

Comment:
I've tried to find documentation on the "near" variable, and failed.

From testing, the near variable seems to be the length of the error, therefor the " " * LENGTH.